File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {

        HashSet<Integer>mapp=new HashSet<>();

        for(int a:nums){
            mapp.add(a);
        }

        ListNode temp=new ListNode(0);
        temp.next=head;
        ListNode node=temp;

        while(node.next!=null){
            if(mapp.contains(node.next.val)){
                node.next=node.next.next;
            }
            else
            node=node.next;
        }
return temp.next;
        
        
    }
}

File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {

                unordered_set<int> mapp(nums.begin(), nums.end());

        ListNode* temp=new ListNode(0);
        temp->next=head;
        ListNode* node=temp;

        while(node->next){
            if(mapp.count(node->next->val)){
                node->next=node->next->next;
            }
            else
            node=node->next;
        }
        return temp->next;
        
    }
};